components/org.wso2.carbon.identity.application.authenticator.basicauth/src/main/java/org/wso2/carbon/identity/application/authenticator/basicauth/BasicAuthenticator.java

```diff
@@ -734,4 +745,40 @@ private Cookie getAutoLoginCookie(Cookie[] cookiesInRequest) {
 
         return targetCookie.orElse(null);
     }
+
+    private boolean isAuthPolicyAccountExistCheck() {
+
+        return Boolean.parseBoolean(IdentityUtil.getProperty(BasicAuthenticatorConstants.AUTHENTICATION_POLICY_CONFIG));
+    }
+
+    /**
+     * Check user existence and set error code to IdentityErrorMsgContext if user does not exist.
+     */
+    private void checkUserExistence() {
+
+        if (!isUserExist()) {
+            IdentityErrorMsgContext customErrorMessageContext = new IdentityErrorMsgContext(UserCoreConstants
+                    .ErrorCode.USER_DOES_NOT_EXIST);
+            IdentityUtil.setIdentityErrorMsg(customErrorMessageContext);
+        }
+    }
+
+    private Boolean isUserExist() {
```

Comment:
I think we can move this content to the other method since there is only one usage? WDYT?

Reply:
As per the code review[1], we decided to add this method.

[1] https://github.com/wso2-support/identity-local-auth-basicauth/pull/28#discussion_r501452252